frontend_tests/node_tests/popover_menus_data.js

```diff
@@ -0,0 +1,248 @@
+    is_admin: false,
+    is_guest: false,
+};
+
+// Helper functions:
+function add_initialize_users() {
+    // Initialize people
+    people.init();
+
+    // Add users
+    people.add_active_user(mike);
+    people.add_active_user(bob);
+    people.add_active_user(me);
+
+    // Initialize current user
+    people.initialize_current_user(me.user_id);
+}
+
+function init_message_list() {
+    const filter = new Filter();
+    const list = new MessageList({
+        filter,
+    });
+
+    assert.equal(list.empty(), true);
+
+    return list;
+}
+
+// Append message to message_list, also add container to message_lists
+function add_message_with_view(list, messages) {
+    list.append(messages, true);
+    for (const message of messages) {
+        message_lists.current.view.message_containers.set(message.id, {
+            is_hidden: message.is_hidden,
+        });
+    }
+}
+
+// Test init function
+function test(label, f) {
+    run_test(label, (helpers) => {
+        // Clear stuff for testing environment
+        add_initialize_users();
+        message_lists.initialize();
+        f(helpers);
+    });
+}
+
+// Test functions
+test("my_message_all_actions", () => {
+    // Set page parameters.
+    page_params.is_spectator = false;
+    page_params.realm_allow_message_editing = true;
+    page_params.realm_message_content_edit_limit_seconds = null;
+    page_params.realm_allow_edit_history = true;
+    page_params.realm_message_content_delete_limit_seconds = null;
+    page_params.realm_delete_own_message_policy = 1;
+    page_params.realm_enable_read_receipts = true;
+
+    // Get message with maximum permissions available
+    // Initialize message list
+    const list = init_message_list();
+    message_lists.set_current(list);
+
+    // Assume message has been previously edited.
+    const messages = [
+        {
+            id: 1,
+            sender_id: me.user_id,
+            is_hidden: false,
+            sent_by_me: true,
+            locally_echoed: false,
+            is_stream: true,
+            stream_id: 1,
+            unread: false,
+            edit_history: [
+                {
+                    prev_content: "Previous content",
+                    prev_stream: 0,
+                    prev_topic: "Previous topic",
+                },
+            ],
+        },
+    ];
+
+    add_message_with_view(list, messages);
+
+    const response = popover_menus_data.get_actions_popover_content_context(1);
+    assert.equal(response.message_id, 1);
+    assert.equal(response.stream_id, 1);
+    assert.equal(response.editability_menu_item, $t({defaultMessage: "Edit message"}));
+    assert.equal(response.move_message_menu_item, $t({defaultMessage: "Move messages"}));
+    assert.equal(response.should_display_mark_as_unread, true);
+    assert.equal(response.view_source_menu_item, undefined);
+    assert.equal(response.should_display_collapse, true);
+    assert.equal(response.should_display_uncollapse, undefined);
```

Comment:
Interesting -- is it a (very minor) logic bug that this is `undefined` rather than `false`?

Reply:
Good catch--it looks like this is indeed a minor logic bug. It seems as though in boolean expressions, ```undefined``` is evaluated by the ```!``` operator to true, however ```undefined``` without the ```!``` operator causes the statement evaluate to ```undefined```.

Because I did not set ```message.collapsed```, this caused line 82 of ```popover_menus_data.js``` to evaluate to undefined instead of false. In my fix I have made sure to set the collapsed field of the message object.